### PR TITLE
Sched-ops changes to support torpedo for k8s 1.7.5

### PR DESCRIPTION
**Pull request by Rohit-PX**

### CODE REVIEW

File: k8s/k8s.go [hunk 1]
```diff
@@ -133,14 +144,23 @@ type StatefulSetOps interface {
...
 	GetStatefulSetsUsingStorageClass(scName string) ([]apps_api.StatefulSet, error)
+	// GetPVCsForStatefulSet returns all the PVCs for given stateful set
+	GetPVCsForStatefulSet(ss *apps_api.StatefulSet) (*v1.PersistentVolumeClaimList, error)
+	// ValidatePVCsForStatefulSet validates the PVCs for the given stateful set
+	ValidatePVCsForStatefulSet(ss *apps_api.StatefulSet) error
 }
 
 // DeploymentOps is an interface to perform k8s deployment operations
 type DeploymentOps interface {
+
```
Comment (piyush-nimbalkar) [inline k8s/k8s.go]: nit: extra empty line

File: k8s/k8s.go [hunk 2]
```diff
@@ -243,7 +278,8 @@ type PersistentVolumeClaimOps interface {
 	// CreatePersistentVolumeClaim creates the given persistent volume claim
 	CreatePersistentVolumeClaim(*v1.PersistentVolumeClaim) (*v1.PersistentVolumeClaim, error)
 	// DeletePersistentVolumeClaim deletes the given persistent volume claim
-	DeletePersistentVolumeClaim(*v1.PersistentVolumeClaim) error
+	DeletePersistentVolumeClaim(name, namespace string) error
+
```
Comment (piyush-nimbalkar) [inline k8s/k8s.go]: nit: extra empty line

File: k8s/k8s.go [hunk 3]
```diff
@@ -1890,6 +2198,26 @@ func (k *k8sOps) UpdateConfigMap(configMap *v1.ConfigMap) (*v1.ConfigMap, error)
...
+// ListEvents retrieves all events registered with kubernetes
+func (k *k8sOps) ListEvents(namespace string, opts meta_v1.ListOptions) (*v1.EventList, error) {
+	if err := k.initK8sClient(); err != nil {
+		return nil, err
+	}
+	return k.client.CoreV1().Events(namespace).List(opts)
+}
+
+//func (k *k8sOps) appsClient() appsv1beta1.AppsV1beta1Interface {
```
Comment (piyush-nimbalkar) [inline k8s/k8s.go]: where ever you are using k.client.Apps(), you should be using this appsClient() method
and here in this method you should use, `return k.client.AppsV1beta1()`

**Comment by Rohit-PX** [comment]
Fixed travis errors.

File: k8s/k8s.go (hunk 1 above)
Comment (Rohit-PX) [reply k8s/k8s.go]: Fixed

**Pull request merged**